lib/maths/analytics/CBoostedTreeHyperparameters.cc

```diff
@@ -687,13 +668,15 @@ std::size_t CBoostedTreeHyperparameters::estimateMemoryUsage() const {
     return sizeof(*this) + numberToTune * sizeof(int) +
            (m_NumberRounds / 3 + 1) * numberToTune * sizeof(double) +
            common::CBayesianOptimisation::estimateMemoryUsage(numberToTune, m_NumberRounds) +
+           numberToTune * sizeof(std::size_t) +
            numberToTune * maxLineSearchIterations() * sizeof(double);
```

Comment:
I am starting to struggle remembering what summand is responsible for what. Could you maybe add an explanation comment?

Reply:
I added extra documentation in [bcdd707](https://github.com/elastic/ml-cpp/pull/2325/commits/bcdd707c74d865ff71ca348ac765e963a5f3b9f6).